lib/rules/jsx-no-update-forcing-props.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * @fileoverview Prevent usage of props that would cause the component to render every time its parent renders
+ * @author Wojciech Maj
+ */
+'use strict';
+
+const propName = require('jsx-ast-utils/propName');
+const docsUrl = require('../util/docsUrl');
+const jsxUtil = require('../util/jsx');
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent usage of props that would cause the component to render every time its parent renders',
+      category: 'Best Practices',
+      recommended: false,
+      url: docsUrl('jsx-no-update-forcing-props')
+    },
+    schema: [{
+      type: 'object',
+      properties: {
+        allowObjectExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        allowArrayExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        allowFunctionExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        allowArrowFunctionExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        allowNewExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        allowCallExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        ignoreRefs: {
+          default: false,
+          type: 'boolean'
+        }
```

Comment:
maybe rather than 7 booleans, 6 of which are named "allow" and default to false, and the 7th named "ignore" and defaults to true - we could use an object like:
```
{
  functions: 'allow',
  arrowFunctions: 'allow',
  objects: 'allow',
  arrays: 'allow',
  newExpressions: 'allow',
  callExpressions: 'allow',
  refs: 'ignore',
}
```
?